coalib/output/ConsoleInteraction.py

```diff
@@ -479,16 +479,20 @@ def require_setting(setting_name, arr):
     :param arr:          A list containing a description in [0] and the name
                          of the bears who need this setting in [1] and
                          following.
+    :param section:      The section the action corresponds to.
     """
     needed = join_names(arr[1:])
+    section_name = "from section " + "\"" + section.name_str + "\"" + ": "
 
     # Don't use input, it can't deal with escapes!
+
     print(colored(STR_GET_VAL_FOR_SETTING.format(setting_name, arr[0], needed),
-                  REQUIRED_SETTINGS_COLOR))
+                  REQUIRED_SETTINGS_COLOR) + colored(section_name,
+                                                     REQUIRED_SETTINGS_COLOR))
```

Comment:
Instead change `STR_GET_VAL_FOR_SETTING` to `"Please enter a value for the setting \"{}\" ({}) needed by {} from section \"{}\":"`. And here, use:

```
print(colored(STR_GET_VAL_FOR_SETTING.format(setting_name, arr[0], needed, section.name_str),
                  REQUIRED_SETTINGS_COLOR))
```

Much much cleaner IMO